source/extensions/filters/http/common/aws/credentials_provider.h

```diff
@@ -25,6 +33,34 @@ class Credentials {
       : access_key_id_(access_key_id), secret_access_key_(secret_access_key),
         session_token_(session_token) {}
 
+  static Credentials fromString(const std::string& access_key_id,
+                                const std::string& secret_access_key,
+                                const std::string& session_token) {
+    if (!(access_key_id.empty() || secret_access_key.empty() || session_token.empty())) {
+      return Credentials(access_key_id, secret_access_key, session_token);
+    } else if (!(access_key_id.empty() || secret_access_key.empty())) {
+      return Credentials(access_key_id, secret_access_key);
+    } else if (!access_key_id.empty()) {
+      return Credentials(access_key_id);
+    }
+
+    return Credentials();
+  }
+
+  static Credentials fromCString(const char* access_key_id, const char* secret_access_key,
```

Comment:
I'm not sure we need to have fromString and fromCString here. Could you explain more why we need those two? What do you think about `string_view`?

Reply:
Replaced `fromString()` and `fromCString()` and all `Credentials` constructors with a single constructor handling all cases.